[SwiftUI] Add more functionality to SwiftBrowser to aid debugging

rr-codes opened the pull request:
#### 09150f408020a7eebfdd37343b14ed647b45a5ab
<pre>
[SwiftUI] Add more functionality to SwiftBrowser to aid debugging
<a href="https://bugs.webkit.org/show_bug.cgi?id=298846">https://bugs.webkit.org/show_bug.cgi?id=298846</a>
<a href="https://rdar.apple.com/160587389">rdar://160587389</a>

Reviewed by Aditya Keerthi.

- Make it possible for SwiftBrowser to toggle the &quot;_editable&quot; property
- Make it possible for SwiftBrowser to toggle Smart Lists enablement
- Adapt SwiftBrowser for Liquid Glass, and make it possible to control the scroll edge effect style
- Make it possible for SwiftBrowser to toggle web preference feature flags
- Refactor SwiftBrowser views for better composition
- Add some icons to menu bar items
- Fix some formatting

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(isEditable):
(smartListsEnabled):
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _isSmartListsEnabled]):
(-[WKWebView _setSmartListsEnabled:]):
(-[WKWebView _toggleSmartLists:]):
(-[WKWebView isSmartListsEnabled]): Deleted.
(-[WKWebView setSmartListsEnabled:]): Deleted.
(-[WKWebView toggleSmartLists:]): Deleted.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::firstRectForCharacterRange):
* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
(View.webViewScrollEdgeEffectStyle(_:for:)):
* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
(WebPreferenceFeature.allowSmartLists):
* Source/WebKit/_WebKit_SwiftUI/Implementation/EnvironmentValues+Extras.swift:
(EnvironmentValues.webViewScrollEdgeEffectStyleContext):
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
(WebPreferenceContext.features):
(WebPreferenceContext.isEmpty):
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:
(WebViewRepresentable.updatePlatformView(_:context:)):
* Tools/SwiftBrowser/Source/Modifiers/WebContextMenu.swift: Added.
(WebContextMenu.body(_:)):
(View.webContextMenu):
* Tools/SwiftBrowser/Source/Modifiers/WebToolbarModifier.swift: Added.
(body):
(LabelConfiguration.body):
(WebToolbarModifier.findNavigatorIsPresented):
(WebToolbarModifier.body(_:)):
(View.webToolbar(_:)):
* Tools/SwiftBrowser/Source/SwiftBrowser.swift:
(SwiftBrowserApp.mostRecentURL):
(SwiftBrowserApp.body):
* Tools/SwiftBrowser/Source/ViewModel/AppStorageKeys.swift:
* Tools/SwiftBrowser/Source/Views/BrowserView.swift:
(BrowserView.url):
(BrowserView.body):
* Tools/SwiftBrowser/Source/Views/ContentView.swift:
(ContentView.url):
(ContentView.scrollBounceBehaviorBasedOnSize):
(ContentView.backgroundHidden):
(ContentView.showColorInTabBar):
(ContentView.body):
(body): Deleted.
(LabelConfiguration.body): Deleted.
(PrincipalToolbarGroup.body): Deleted.
* Tools/SwiftBrowser/Source/Views/SettingsView.swift:
(PermissionDecisionView.permissionDecision):
(BinaryValuePicker.value):
(GeneralSettingsView.body):
* Tools/SwiftBrowser/SwiftBrowser.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SmartLists.mm:
(TEST(SmartLists, EnablementIsLogicallyConsistentWhenInterfacedThroughResponder)):

Canonical link: <a href="https://commits.webkit.org/299974@main">https://commits.webkit.org/299974@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5397612e1b3a515834029cbecc81ee621bbcf860

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120918 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40612 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31268 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127328 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72994 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6e19f357-f315-4931-9d55-91b65a147b2a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122794 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41310 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49189 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91836 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61101 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/df818df1-f818-4891-9565-efd463357edb) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123870 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32987 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108400 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72531 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/13c14d18-3aee-43f9-aedb-8268cce2bf5b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32017 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26502 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70920 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102483 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26681 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130186 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47839 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36341 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100453 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48208 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104573 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100356 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25442 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45764 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23813 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44530 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47701 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53406 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47172 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50516 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48856 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2ac5dd49a83db3119f6caa3c137792a61a588ccc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50729

rr-codes committed:
[SwiftUI] Add more functionality to SwiftBrowser to aid debugging
https://bugs.webkit.org/show_bug.cgi?id=298846
rdar://160587389

Reviewed by Aditya Keerthi.

- Make it possible for SwiftBrowser to toggle the "_editable" property
- Make it possible for SwiftBrowser to toggle Smart Lists enablement
- Adapt SwiftBrowser for Liquid Glass, and make it possible to control the scroll edge effect style
- Make it possible for SwiftBrowser to toggle web preference feature flags
- Refactor SwiftBrowser views for better composition
- Add some icons to menu bar items
- Fix some formatting

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(isEditable):
(smartListsEnabled):
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _isSmartListsEnabled]):
(-[WKWebView _setSmartListsEnabled:]):
(-[WKWebView _toggleSmartLists:]):
(-[WKWebView isSmartListsEnabled]): Deleted.
(-[WKWebView setSmartListsEnabled:]): Deleted.
(-[WKWebView toggleSmartLists:]): Deleted.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::firstRectForCharacterRange):
* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
(View.webViewScrollEdgeEffectStyle(_:for:)):
* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
(WebPreferenceFeature.allowSmartLists):
* Source/WebKit/_WebKit_SwiftUI/Implementation/EnvironmentValues+Extras.swift:
(EnvironmentValues.webViewScrollEdgeEffectStyleContext):
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
(WebPreferenceContext.features):
(WebPreferenceContext.isEmpty):
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:
(WebViewRepresentable.updatePlatformView(_:context:)):
* Tools/SwiftBrowser/Source/Modifiers/WebContextMenu.swift: Added.
(WebContextMenu.body(_:)):
(View.webContextMenu):
* Tools/SwiftBrowser/Source/Modifiers/WebToolbarModifier.swift: Added.
(body):
(LabelConfiguration.body):
(WebToolbarModifier.findNavigatorIsPresented):
(WebToolbarModifier.body(_:)):
(View.webToolbar(_:)):
* Tools/SwiftBrowser/Source/SwiftBrowser.swift:
(SwiftBrowserApp.mostRecentURL):
(SwiftBrowserApp.body):
* Tools/SwiftBrowser/Source/ViewModel/AppStorageKeys.swift:
* Tools/SwiftBrowser/Source/Views/BrowserView.swift:
(BrowserView.url):
(BrowserView.body):
* Tools/SwiftBrowser/Source/Views/ContentView.swift:
(ContentView.url):
(ContentView.scrollBounceBehaviorBasedOnSize):
(ContentView.backgroundHidden):
(ContentView.showColorInTabBar):
(ContentView.body):
(body): Deleted.
(LabelConfiguration.body): Deleted.
(PrincipalToolbarGroup.body): Deleted.
* Tools/SwiftBrowser/Source/Views/SettingsView.swift:
(PermissionDecisionView.permissionDecision):
(BinaryValuePicker.value):
(GeneralSettingsView.body):
* Tools/SwiftBrowser/SwiftBrowser.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SmartLists.mm:
(TEST(SmartLists, EnablementIsLogicallyConsistentWhenInterfacedThroughResponder)):

Canonical link: https://commits.webkit.org/299974@main
